google/cloud/bigquery/_tqdm_helpers.py

```diff
@@ -53,16 +58,34 @@ def get_progress_bar(progress_bar_type, description, total, unit):
     return None
 
 
-def wait_for_query(query_job, progress_bar_type=None):
-    """Return query result and display a progress bar while the query running, if tqdm is installed."""
+def wait_for_query(
+    query_job: "QueryJob",
+    progress_bar_type: Optional[str] = None,
+    max_results: Optional[int] = None,
+) -> Any:
```

Comment:
The return type annotation was set to `Any`, because when set to `Union["RowIterator", "_EmptyRowIterator"]`, the `pytype` check failed with the following error:

```
File "/tmpfs/src/github/python-bigquery/google/cloud/bigquery/_tqdm_helpers.py", line 63, in <module>: Function dict.__setitem__ was called with the wrong arguments [wrong-arg-types]
         Expected: (self: Dict[str, Any], ...)
  Actually passed: (self: Dict[str, Union[Type[Optional[int]], str]], ...)
```

This is a bit odd, because line 63 used to be the line where `wait_for_query()` was defined, i.e. `def wait_for_query(`. It's not clear to me what dicts `pytype` complained about. :confused:

Reply:
Maybe try blowing away the `.pytype/` directory?

Reply:
Did that, same result. And the local error was consistent with what failed on Kokoro, too.

Reply:
Is there a base class for `RowIterator` / `_EmptyRowIterator` which could be used?  Ideally, the caller shouldn't care about the "empty" iterator:  it should just behave normally.

Reply:
There isn't. `RowIterator` inherits from `HTTPIterator`, while `_EmptyRowIterator` directly subclasses object.

I agree that users shouldn't care, however. After all both are the same thing, in principle, except that one's result set is empty. I'll see if changing the `_EmptyRowIterator`'s base class works.

Reply:
**Update:** Making `_EmptyRowIterator` a subclass of `RowIterator` (with some plumbing) seems to work, setting `RowIterator` as the return type now passes the check.

I'll see if there's anything else to override in the parent iterator class, but this approach seems promising.